### PR TITLE
DEVEX-638 Xenial Asset Builder

### PR DESCRIPTION
**Pull request by geetduggal**
WIP Initial commit;  should land with: https://github.com/dnanexus/dx_app_builder/pull/13

### CODE REVIEW

File: src/python/dxpy/asset_builder.py [hunk 1]
```diff
@@ -78,9 +79,10 @@ def validate_conf(asset_conf):
     """
     if 'name' not in asset_conf:
         raise AssetBuilderException('The asset configuration does not contain the required field "name".')
+    # TODO: this default is not a good idea, and we will have to remove it once we ask customers to always provide release
     if 'release' not in asset_conf:
         asset_conf['release'] = "12.04"
-    elif asset_conf['release'] != '12.04' and asset_conf['release'] != '14.04':
+    elif asset_conf['release'] not in ['12.04', '14.04', '16.04']:
         raise AssetBuilderException('The "release" field value should be either "12.04" or "14.04".')
```
Comment (commandlinegirl) [inline src/python/dxpy/asset_builder.py]: Please, add 16.04 to the error message.

Comment (commandlinegirl) [reply src/python/dxpy/asset_builder.py]: And maybe the order should be: "16.04, 14.04, 12.04 (deprecated)"

File: contrib/perl/lib/DNAnexus/API.pm [hunk 2]
```diff
@@ -1342,4 +1342,5 @@ sub workflowNew(;$%) {
...
...jects projectRemoveTags projectRenameFolder projectSetProperties projectTransfer projectUpdate projectUpdateSponsorship projectNew recordAddTags recordAddTypes recordClose recordDescribe recordGetDetails recordListProjects recordRemoveTags recordRemoveTypes recordRename recordSetDetails recordSetProperties recordSetVisibility recordNew systemDescribeDataObjects systemDescribeExecutions systemDescribeProjects systemFindAffiliates systemFindApps systemFindDataObjects systemResolveDataObjects systemFindExecutions systemFindAnalyses systemFindJobs systemFindProjects systemFindUsers systemFindProjectMembers systemFindOrgs systemGenerateBatchInputs systemGlobalSearch systemGreet systemHeaders systemShortenURL systemWhoami userDescribe userUpdate workflowAddStage workflowAddTags workflowAddTypes workflowClose workflowDescribe workflowDryRun workflowGetDetails workflowIsStageCompatible workflowListProjects workflowMoveStage workflowOverwrite workflowRemoveStage workflowRemoveTags workflowRemoveTypes workflowRename workflowRun workflowValidateBatch workflowSetDetails workflowSetProperties workflowSetStageInputs workflowSetVisibility workflowUpdate workflowUpdateStageExecutable workflowNew);
```
Comment (commandlinegirl) [inline contrib/perl/lib/DNAnexus/API.pm]: What changes in this line?

Comment (commandlinegirl) [reply contrib/perl/lib/DNAnexus/API.pm]: Oh, it's systemGenerateBatchInputs, do we need it in this commit?

Comment (geetduggal) [reply contrib/perl/lib/DNAnexus/API.pm]: Oh that's weird -- don't know how that got in there.  I"ll remove it

**Pull request merged**